nntrainer/graph/network_graph.cpp

```diff
@@ -494,6 +511,47 @@ void NetworkGraph::backwarding(
     backwarding_op(ln, iteration);
     END_PROFILE(profile_keys.at(ln->getType()));
   }
+
+  /** perform clipping of the gradients by global norm if any */
+  auto clip_weights = tensor_manager->getWeights([](Weight *w) {
+    return w->hasGradient() && w->isGradientLastAccess() &&
+           w->isGradientClipByGlobalNorm();
+  });
```

Comment:
nitpicking: can we cache this vector as a member to avoid this calculation every iteration?

Reply:
will add.